### PR TITLE
Secret Manager Script

### PR DESCRIPTION
**Pull request by DGieseke**
This PR introduces a new `secrets.py` file under the `scrapers/utils` directory, alongside other shared utility files. This file contains a `get_secret` function, which enables retrieving secrets from AWS Secrets Manager.

Additionally, the `NEW_YORK_API_KEY` in the NY scraper has been updated to use the new get_secret function, pulling the key directly from AWS Secrets Manager instead of relying on environment variables.

### CODE REVIEW

File: scrapers/utils/secrets.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    Returns:
+    - The value associated with the specified key.
+    """
+
+    client = boto3.client(service_name="secretsmanager", region_name=region)
+
+    try:
+        secret_response = client.get_secret_value(SecretId=secret_name)
+
```
Comment (jessicabowden) [inline scrapers/utils/secrets.py:21]: nit: remove whitespace here

File: scrapers/utils/secrets.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    except Exception as e:
+        print(f"Error retrieving secret: {e}")
```
Comment (jessicabowden) [inline scrapers/utils/secrets.py:23]: is there no existing logger?

**Review by jessicabowden — CHANGES_REQUESTED** [review]

File: scrapers/utils/secrets.py (hunk 1 above)
Comment (DGieseke) [reply scrapers/utils/secrets.py]: Removed!

File: scrapers/utils/secrets.py (hunk 2 above)
Comment (DGieseke) [reply scrapers/utils/secrets.py:22]: It looks like the openstates-core and openstates-scrapers repos use the `logging` library - if it makes sense to you, I can remove the print statements and add in `logging` warnings instead!

Comment (DGieseke) [reply scrapers/utils/secrets.py]: UPDATE: I added in `logging` warnings in lieu of print statements.

File: scrapers/utils/secrets.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - The value associated with the specified key.
+    """
+
+    client = boto3.client(service_name="secretsmanager", region_name=region)
+
+    try:
+        secret_response = client.get_secret_value(SecretId=secret_name)
+    except Exception as e:
+        logging.warning(f"Error retrieving secret: {e}")
+        return None
```
Comment (jessicabowden) [inline scrapers/utils/secrets.py:24]: What's going to happen if this is `None`? Think about this both in relation to the update in this PR but also elsewhere we're going to use this

Comment (DGieseke) [reply scrapers/utils/secrets.py]: I changed this script to raise an exception to stop the script at this point, but please let me know if I misunderstood or if this is too duplicative (raising an exception as well as logging an error).

File: scrapers/utils/secrets.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - secret_name (str): The name of the secret.
+    - region (str): The AWS region where the secret is stored.
+
+    Returns:
+    - The value associated with the specified key.
+    """
+
+    client = boto3.client(service_name="secretsmanager", region_name=region)
+
+    try:
```
Comment (jessicabowden) [inline scrapers/utils/secrets.py:34]: This is looking better but we've got a lot of repetition now with the error handling.

We could simplify this to:
```
try:
    secret_response = client.get_secret_value(SecretId=secret_name)
    secret_string = secret_response.get("SecretString")
    if not secret_string:
        raise SecretRetrievalError("Secret is binary or unavailable in string format", secret_name)

    secret = json.loads(secret_string).get(secret_name)
    if not secret:
        raise SecretRetrievalError("The key was not found in the secret", secret_name)

    return secret

except Exception as e:
    raise SecretRetrievalError("Error retrieving secret", secret_name, e)

```

Comment (DGieseke) [reply scrapers/utils/secrets.py:34]: Thanks for the suggestion! I incorporated this change into my code, tested it, reformatted it, and it works perfectly.

**Review by jessicabowden — APPROVED** [review]
Commit history is really messy for such a small PR - please clean up before merging

**Pull request merged**